packages/big-design/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,100 @@
+import { ArrowDropDownIcon, ChevronLeftIcon, ChevronRightIcon } from '@bigcommerce/big-design-icons';
+import React, { useEffect, useState } from 'react';
+
+import { MarginProps } from '../../mixins';
+import { Box } from '../Box';
+import { Dropdown } from '../Dropdown';
+import { Flex } from '../Flex';
+
+import { StyledButton, StyledIconButton } from './styled';
+
+export interface PaginationProps extends MarginProps {
+  currentRange: number;
+  currentPage: number;
+  totalItems: number;
+  rangeOptions: number[];
+  onPageChange(page: number): void;
+  onRangeChange(range: number): void;
+}
+
+export const Pagination: React.FC<PaginationProps> = props => {
+  const { currentRange, currentPage, totalItems, rangeOptions, onPageChange, onRangeChange } = props;
+
+  const [maxPages, setMaxPages] = useState(Math.ceil(totalItems / currentRange));
+  const [itemRange, setItemRange] = useState([0, 0]);
+
+  useEffect(() => {
+    const firstItemInRange = currentRange * (currentPage - 1) + 1;
+    let lastItemInRange = currentRange * currentPage;
+    if (lastItemInRange > totalItems) {
+      lastItemInRange = totalItems;
+    }
+
+    setItemRange([firstItemInRange, lastItemInRange]);
+  }, [currentPage, currentRange, totalItems]);
```

Comment:
Correct me if I'm wrong, but if `totalItems` changes, `maxPages` will not update. So with that in mind, you'll maybe want to have another `useEffect` listening to only `totalItems` and update it whenever that changes.

Reply:
Since the state of `currentPage` is dependent on `totalItems` and `currentRange` props, it should/does change dynamically without the need for a `useEffect` listener (at least, that has been my experience with the component). We may need to look at messing around with this to be certain though.